app/code/Magento/Newsletter/Controller/Manage/Save.php

```diff
@@ -74,13 +76,28 @@ public function execute()
                 $customer = $this->customerRepository->getById($customerId);
                 $storeId = $this->storeManager->getStore()->getId();
                 $customer->setStoreId($storeId);
-                $this->customerRepository->save($customer);
-                if ((boolean)$this->getRequest()->getParam('is_subscribed', false)) {
-                    $this->subscriberFactory->create()->subscribeCustomerById($customerId);
-                    $this->messageManager->addSuccess(__('We saved the subscription.'));
+                $isSubscribedState = $customer->getExtensionAttributes()
+                    ->getIsSubscribed();
+                $isSubscribedParam = (boolean)$this->getRequest()
+                    ->getParam('is_subscribed', false);
+                if ($isSubscribedParam != $isSubscribedState) {
+                    $this->customerRepository->save($customer);
+                    if ($isSubscribedParam) {
+                        $subscribeModel = $this->subscriberFactory->create()
+                            ->subscribeCustomerById($customerId);
+                        $subscribeStatus = $subscribeModel->getStatus();
+                        if ($subscribeStatus == Subscriber::STATUS_SUBSCRIBED) {
+                            $this->messageManager->addSuccess(__('We saved the subscription.'));
+                        } else {
+                            $this->messageManager->addSuccess(__('The confirmation request has been sent.'));
+                        }
+                    } else {
+                        $this->subscriberFactory->create()
+                            ->unsubscribeCustomerById($customerId);
+                        $this->messageManager->addSuccess(__('We removed the subscription.'));
+                    }
                 } else {
-                    $this->subscriberFactory->create()->unsubscribeCustomerById($customerId);
-                    $this->messageManager->addSuccess(__('We removed the subscription.'));
+                    $this->messageManager->addSuccess(__('We updated the subscription.'));
```

Comment:
Again to make it more personal to the user "We have updated your subscription" would be good here.